### ISSUE TITLE
AllowReEntry not working

### ISSUE THREAD

**Issue by bazuker** [PaulSonOfLars/gotgbot#91]
I am having trouble with `AllowReEntry` in `ConversationOpts`. 
```Go
func GetConversationStarters() []ext.Handler {
	return []ext.Handler{
		handlers.NewCommand("start", start),
	}
...
dispatcher.AddHandler(handlers.NewConversation(
GetConversationStarters(),
GetConversationHandlers(),
&handlers.ConversationOpts{
	Exits:        []ext.Handler{handlers.NewCommand("cancel", cancel)},
	...
	AllowReEntry: true,
     },
))
```

It is set to `true`, however I can't get it to work and the conversation does not restart.
From my expectations, `/start` command should be working regardless of the current state.

**Comment by PaulSonOfLars** [comment]
Hi Daniil, thanks for raising the issue!

That's very strange, I've got a test to check specifically for this behaviour. 

Would you be able to provide us with a minimal reproduction of this, so we can test against it and reproduce?

**Comment by PaulSonOfLars** [comment]
Ah, managed to repro locally, the test was wrong - Fixed in v2.0.0-rc.20!

**Comment by bazuker** [comment]
@PaulSonOfLars thanks for a quick fix
